qsym/pintool/solver.cpp

```diff
@@ -510,6 +512,11 @@ ExprRef Solver::getRangeConstraint(ExprRef e, bool is_unsigned) {
 
 bool Solver::isInterestingJcc(ExprRef rel_expr, bool taken, ADDRINT pc) {
   bool interesting = trace_.isInterestingBranch(pc, taken);
+
+  // The block must meet 2 condition: 1)never solved before 2)is state-related.
+  if (interesting && cfg_.isStateRelated(cur_bb_first_addr_)) interesting = true;
```

Comment:
What if the current branch taken is right? Reverting it may lead to a path NOT towards stateful node.